### PR TITLE
Feature/student timeline

### PR DESCRIPTION
**Pull request by norberthcsorba**

### CODE REVIEW

File: src/main/resources/static/js/professor/timeline.js [hunk 1]
```diff
@@ -177,55 +84,53 @@ function assignLessonCellHandlers(enrlRow, lessonCell) {
...
-                case "LABORATORY":
-                    nrOfLabAttendances++;
-                    break;
-                case "SEMINAR":
-                    nrOfSemAttendances++;
-                    break;
+function assignActionHandlers() {
+    $("#btn_submitChanges").click(() => {
+        let reqBody = JSON.stringify(stripReadOnlyLessons(getLessonList()));
+        console.log(reqBody);
```
Comment (albuoana) [inline src/main/resources/static/js/professor/timeline.js]: -delete console

File: src/main/resources/static/js/student/timeline.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+function updateTotalAttendanceView(enrlRow, totalLabAttendance, totalSemAttendance) {
+    let totalAttendanceCell = $(enrlRow).children("td.attendances-cell").first();
+    let precisionLab = (totalLabAttendance < 100 ? 2 : 0), precisionSem = (totalSemAttendance < 100 ? 2 : 0);
+    debugger;
+    let totalAttendanceViewText;
+    totalAttendanceViewText = `${totalSemAttendance.toFixed(precisionSem)}% / ${totalLabAttendance.toFixed(precisionLab)}%`;
+    console.log(totalAttendanceViewText)
```
Comment (albuoana) [inline src/main/resources/static/js/student/timeline.js]: delete console

File: src/main/resources/static/js/student/timeline.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+function updateTotalAttendanceView(enrlRow, totalLabAttendance, totalSemAttendance) {
+    let totalAttendanceCell = $(enrlRow).children("td.attendances-cell").first();
+    let precisionLab = (totalLabAttendance < 100 ? 2 : 0), precisionSem = (totalSemAttendance < 100 ? 2 : 0);
+    debugger;
```
Comment (albuoana) [inline src/main/resources/static/js/student/timeline.js]: delete debugger

**Review by albuoana — CHANGES_REQUESTED** [review]

File: src/main/java/bussiness_layer/services/IStudentService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package bussiness_layer.services;
+
+import java.util.List;
+
+import bussiness_layer.dto.EnrollmentDto;
+import bussiness_layer.dto.TeachingDto;
+
+public interface IStudentService {
+    List<EnrollmentDto> getEnrollments(String studUsername);
```
Comment (JohnPaul97) [inline src/main/java/bussiness_layer/services/IStudentService.java:14]: Add java docs here

File: src/main/java/bussiness_layer/services/StudentService.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    private IEnrollmentRepository enrollmentRepository;
+
+
+    @Override
+    public List<EnrollmentDto> getEnrollments(String studUsername) {
+        List<Enrollment> enrollments = enrollmentRepository.findByStudent(studUsername);
+        if (enrollments.size() == 0) {
+            throw new ResourceNotFoundException();
+        }
+        List<EnrollmentDto> enrollmentDTOS = EnrollmentMapper.toEntityDtoList(enrollments);
```
Comment (JohnPaul97) [inline src/main/java/bussiness_layer/services/StudentService.java]: rename variable to enrollmentDtos

File: src/main/java/web_layer/controllers/StudentMvcController.java [hunk 6]
```diff
@@ -1,16 +1,36 @@
...
+    @Autowired
+    private IStudentService service;
 
     @GetMapping("/timeline")
-    public ModelAndView getTimelinePage() {
-        return new ModelAndView("/student/timeline");
+    public ModelAndView getTimelinePage(Principal crtUser) {
+        ModelAndView mv = new ModelAndView("/student/timeline");
+        List<EnrollmentDto> enrollments = service.getEnrollments(crtUser.getName());
+        enrollments.forEach(x-> System.out.println(x.getStudent().getUsername() +" "+
```
Comment (JohnPaul97) [inline src/main/java/web_layer/controllers/StudentMvcController.java]: Remove this prints please.

File: src/main/java/bussiness_layer/services/StudentService.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+@Service
+@Transactional
+public class StudentService implements IStudentService {
+
+    @Autowired
+    private IEnrollmentRepository enrollmentRepository;
+
+
+    @Override
+    public List<EnrollmentDto> getEnrollments(String studUsername) {
```
Comment (JohnPaul97) [inline src/main/java/bussiness_layer/services/StudentService.java:22]: Write at least two test methods for this function, one for the happy-flow and one for checking if that exception is thrown. Create a StudentIT class in the test ->business_layer->integration

**Pull request merged**